controller/volume_controller.go

```diff
@@ -648,8 +652,8 @@ func (vc *VolumeController) cleanupFailedToScheduledReplicas(v *longhorn.Volume,
 	if healthyCount >= v.Spec.NumberOfReplicas {
 		for _, r := range rs {
 			if !hasEvictionRequestedReplicas {
-				if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" &&
-					r.Spec.NodeID == "" {
+				if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == "" &&
+					(isDataLocalityDisabled(v) || r.Spec.HardNodeAffinity != e.Spec.NodeID) {
```

Comment:
Can we directly use `r.Spec.HardNodeAffinity != v.Status.CurrentNodeID` here?

Reply:
Yes, we can. Fixed. Thank you